pkg/apis/pingcap/v1alpha1/validation/validation.go

```diff
@@ -146,65 +146,95 @@ func validateTiCDCSpec(spec *v1alpha1.TiCDCSpec, fldPath *field.Path) field.Erro
+				))
 			}
-			if config.CommonConfig.Flash.FlashProxy != nil {
-				if config.CommonConfig.Flash.FlashProxy.LogFile != nil {
-					splitPath := strings.Split(*config.CommonConfig.Flash.FlashProxy.LogFile, string(os.PathSeparator))
-					// The log path should be at least /dir/base.log
-					if len(splitPath) < 3 {
-						allErrs = append(allErrs, field.Invalid(path.Child("config.config.flash.flash_proxy.log-file"),
-							config.CommonConfig.Flash.FlashProxy.LogFile,
-							"log path should include at least one level dir."))
-					}
+		}
+
+		if v := config.Common.Get("flash.proxy.log-file"); v != nil {
+			if value, err := v.AsString(); err == nil {
+				splitPath := strings.Split(value, string(os.PathSeparator))
+				// The log path should be at least /dir/base.log
+				if len(splitPath) < 3 {
+					allErrs = append(allErrs, field.Invalid(path.Child("config.config.flash.proxy.log-file"),
+						value,
+						"log path should include at least one level dir."))
 				}
+			} else {
+				allErrs = append(allErrs, field.Invalid(path.Child("config.config.flash.proxy.log-file"),
+					v.Interface(),
+					fmt.Sprintf("should be string type, but is: %v", reflect.TypeOf(v.Interface())),
+				))
 			}
 		}
-		if config.CommonConfig.FlashLogger != nil {
-			if config.CommonConfig.FlashLogger.ServerLog != nil {
-				splitPath := strings.Split(*config.CommonConfig.FlashLogger.ServerLog, string(os.PathSeparator))
+		if v := config.Common.Get("logger.log"); v != nil {
+			if value, err := v.AsString(); err == nil {
+				splitPath := strings.Split(value, string(os.PathSeparator))
 				// The log path should be at least /dir/base.log
 				if len(splitPath) < 3 {
 					allErrs = append(allErrs, field.Invalid(path.Child("config.config.logger.log"),
-						config.CommonConfig.FlashLogger.ServerLog,
+						value,
 						"log path should include at least one level dir."))
 				}
+			} else {
+				allErrs = append(allErrs, field.Invalid(path.Child("config.config.logger.log"),
+					v.Interface(),
+					fmt.Sprintf("should be string type, but is: %v", reflect.TypeOf(v.Interface())),
+				))
 			}
-			if config.CommonConfig.FlashLogger.ErrorLog != nil {
-				splitPath := strings.Split(*config.CommonConfig.FlashLogger.ErrorLog, string(os.PathSeparator))
+		}
+		if v := config.Common.Get("logger.errorlog"); v != nil {
+			if value, err := v.AsString(); err == nil {
+				splitPath := strings.Split(value, string(os.PathSeparator))
 				// The log path should be at least /dir/base.log
 				if len(splitPath) < 3 {
 					allErrs = append(allErrs, field.Invalid(path.Child("config.config.logger.errorlog"),
-						config.CommonConfig.FlashLogger.ErrorLog,
+						value,
 						"log path should include at least one level dir."))
 				}
+			} else {
+				allErrs = append(allErrs, field.Invalid(path.Child("config.config.logger.errorlog"),
+					v.Interface(),
+					fmt.Sprintf("should be string type, but is: %v", reflect.TypeOf(v.Interface())),
+				))
```

Comment:
L171-L233 can be simplied to:
```go
var fields = []string {
    "flash.flash_cluster.log",
    "flash.proxy.log-file",
    "logger.log",
    "logger.errorlog",
}
for _, field := range fields {
    if value, err := v.AsString(); err == nil {
        splitPath := strings.Split(value, string(os.PathSeparator))
        // The log path should be at least /dir/base.log
        if len(splitPath) < 3 {
            allErrs = append(allErrs, field.Invalid(path.Child(field),
                value,
                "log path should include at least one level dir."))
        }
    } else {
        allErrs = append(allErrs, field.Invalid(path.Child(field),
            v.Interface(),
            fmt.Sprintf("should be string type, but is: %v", reflect.TypeOf(v.Interface())),
        ))
    }
}
```

Reply:
https://github.com/pingcap/tidb-operator/pull/3355/commits/1d8257d42f27367e5c219c403c309c18d0a0c2a9